composer/utils/object_store/mlflow_object_store.py

```diff
@@ -62,9 +61,19 @@ def _wrap_mlflow_exceptions(uri: str, e: Exception):
     retryable_client_codes = [ErrorCode.Name(code) for code in [ABORTED, REQUEST_LIMIT_EXCEEDED, RESOURCE_EXHAUSTED]]
     not_found_codes = [ErrorCode.Name(code) for code in [RESOURCE_DOES_NOT_EXIST, NOT_FOUND, ENDPOINT_NOT_FOUND]]
 
+    # MLflow wraps Azure data exceptions as INTERNAL_ERROR. Need to unwrap and check msg for the specific error.
+    non_retryable_internal_error_codes = [
+        '401',
+        '403',
+    ]
+
     if isinstance(e, MlflowException):
         error_code = e.error_code  # pyright: ignore
-        if error_code in retryable_server_codes or error_code in retryable_client_codes:
+        if error_code == ErrorCode.Name(INTERNAL_ERROR):
+            error_message = e.message # pyright: ignore
+            if any(code in error_message for code in non_retryable_internal_error_codes):
```

Comment:
I don't know if this is safe from false positives because the error message contains a long URL with many numbers in it.
```
requests.exceptions.HTTPError: 403 Client Error: This request is not authorized to perform this operation. for url: https:<long url with lots of numbers in it>
```

Reply:
```suggestion
            if any(f'{code} Client Error' in error_message for code in non_retryable_internal_error_codes):
```